crates/containerd-shim-wasmtime/src/instance.rs

```diff
@@ -302,9 +306,32 @@ where
         log::debug!("loading wasm component");
 
         wasmtime_wasi::runtime::in_tokio(async move {
-            tokio::select! {
-                status = self.execute_component_async(ctx, component, func, stdio) => { status }
-                sig = wait_for_signal() => { sig }
+            let mut force_shutdown = false;
+            let exec = self.execute_component_async(ctx, component, func, stdio);
+
+            tokio::pin!(exec);
+
+            loop {
```

Comment:
**comment:** I got a bit confused by this being a loop, but then realized this will only run one, and at most two iterations. I wonder if this can be rewritten without a loop.

Reply:
I think the problem is that we want to continue with server future after we receive a first signal. I 'm not sure if it's possible without loop.

Reply:
Can we do this to make sure that it at most loops twice? 
```rust
for _ in 0..2 {
  ...
}
```

Reply:
It is now rewritten without a loop. Thanks @jprendes for suggestion.